src/components/common/Members/filterMembers.ts

```diff
@@ -0,0 +1,55 @@
+import { FormValues } from '~common/ColonyMembers/MembersFilter';
+import { VerificationType } from '~common/ColonyMembers/MembersFilter/filtersConfig';
+
+import { Contributor, Watcher } from '~types';
+
+export const filterMembers = <M extends Contributor | Watcher>(
+  data: M[],
+  searchValue?: string,
+  filters?: FormValues,
+): M[] => {
+  /* No filters */
+  if (!searchValue && filters?.verificationType === VerificationType.All) {
+    return data;
+  }
+
+  /* Only text filter */
+  if (searchValue && filters?.verificationType === VerificationType.All) {
+    return data.filter(
+      ({ user }) =>
+        user?.profile?.displayName
+          ?.toLowerCase()
+          .includes(searchValue.toLowerCase()) ||
+        user?.walletAddress
+          ?.toLowerCase()
+          .includes(searchValue.toLowerCase()) ||
+        user?.name.toLowerCase().includes(searchValue.toLowerCase()),
+    );
+  }
+
+  /* All other combinations */
+  return data.filter(({ user }) => {
+    const textFilter =
+      searchValue === undefined || searchValue === ''
+        ? true
+        : user?.profile?.displayName
+            ?.toLowerCase()
+            .includes(searchValue.toLowerCase()) ||
+          user?.walletAddress
+            ?.toLowerCase()
+            .includes(searchValue.toLowerCase()) ||
+          user?.name.toLowerCase().includes(searchValue.toLowerCase());
+
+    // if (filters?.verificationType === VerificationType.VERIFIED) {
+    //   return isWhitelisted && textFilter;
+    // }
+
+    // if (filters?.verificationType === VerificationType.UNVERIFIED) {
+    //   return !isWhitelisted && textFilter;
+    // }
+
+    // return !isWhitelisted && textFilter;
+
+    return textFilter;
+  });
```

Comment:
I know you haven't came up with this initially, but why are we chaining things like that?
I think we could split the particular filters (search, verification type) into own small helper functions and then this entire long function could become something like:
```js
members.filter(member => filterMemberBySearchTerm(member, searchTerm) && filterMemberByVerificationStatus(member, verificationStatus))
```
This could be more sophisticated, if needed, but at least we won't be repeating so much code unnecessarily.

Reply:
@jakubcolony split up the logic as suggested